### DISCUSSION TITLE
[QUESTION] volume degrade on same node

### DISCUSSION THREAD

**Discussion by iamatulsingh** [longhorn/longhorn#6885] [General]
## Environment

 - Longhorn version: v1.5.1
 - Kubernetes version: 1.27
 - Node config
   - OS type and version Raspbian 
   - Kernel version 6.1.21
   - CPU per node: 4
   - Memory per node: 20 Gi
   - Disk type hdd
   - Network bandwidth and latency between the nodes:
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): baremetal (Raspberry pi 4 8gb RAM)

## Additional context

I have k3s running on my Raspberry pi 4 (8gb memory and only 1 node cluster) . Whenever I'm trying to create a new volume with replicas more than 1, only 1 replicas is getting created and others are degraded and not creating. 

I have checked the cpu usage bit it was 20% only. 
Is it possible to have replicas on 1 node?

**Comment by DamiaSan** [comment]
> Is it possible to have replicas on 1 node?

Yes, you can have more than one replica per node.

**Reply by iamatulsingh** [reply]
But it's not creating more than 1. Other replicas immediately went to degraded once creating replicas.

**Comment by DamiaSan** [comment]
Maybe the problem is different, can you share additional info for example k3s and longhorn configuration?

**Reply by iamatulsingh** [reply]
I have followed two different blogs. 
1. https://rpi4cluster.com/k3s/k3s-kube-setting/ - installing K3s but only followed till using curl to install K3s. No taint has been used as I only have 1 node which is my RPi4

2. https://gdha.github.io/pi-stories/pi-stories9/  - installation of longhorn.

I have made no special change other than the steps written in those blogs.

Some information:
Raspberry Pi 4 (8Gb RAM)
CPU usage isn't high at all when performing reolicas to create.

**Comment by DamiaSan** [comment]
I don't know those blogs, but maybe you can have a look at the official documentation of [k3s](https://docs.k3s.io/quick-start) and [Longhorn](https://longhorn.io/docs/1.5.1/deploy/install/).
Otherwise more info are needed to understand the problem.

**Reply by iamatulsingh** [reply]
K3s
* I have K3s installed on a Raspberry pi 4 with open iscsi installed so that longhorn can use it.
* K3s currently have traefik installed ingress with metallb
* RPi4 has 8 gb RAM and 20Gb space (schedulable storage)
* Running only 1 node cluster with no taint so that any other pods can scheduled

Longhorn
* installed with csi provisioner
* have attached a load balancer (traefik) for the Frontened


Do you need anything else?

**Reply by DamiaSan** [reply]
Yes, logs and configuration of your environment are needed to understand what is happening. You can generate and share a support-bundle to have a look.

**Reply by iamatulsingh** [reply]
[supportbundle_e69ec67e-9054-430a-8783-13296164df30_2023-10-19T09-46-23Z.zip](https://github.com/longhorn/longhorn/files/13040936/supportbundle_e69ec67e-9054-430a-8783-13296164df30_2023-10-19T09-46-23Z.zip)


Here is my support-bundle exported from longhorn.

**Comment by DamiaSan** [comment]
Created issue #6925

**Comment by derekbit** [comment]
@iamatulsingh
Do you mean on the replicas on **different disks of a node** or **on the same disk of a node**?

**Reply by iamatulsingh** [reply]
@derekbit On the same disk of a node.

**Reply by iamatulsingh** [reply]
I even tried with the different disk on same node (ext4) with almost 900G space left but the issue is same.
